src/powerquery-formatter/themes/themes.ts

```diff
@@ -0,0 +1,389 @@
+         */
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        public readonly parameters?: Record<string, any>,
+    ) {}
+}
+
+/**
+ * Parse a raw theme into first-stage ParsedThemeRule.
+ */
+export function parseTheme(source?: IRawTheme): ParsedThemeRule[] {
+    if (!source) {
+        return [];
+    }
+
+    if (!source.settings || !Array.isArray(source.settings)) {
+        return [];
+    }
+
+    const settings: IRawThemeSetting[] = source.settings;
+    const result: ParsedThemeRule[] = [];
+    let resultLen: number = 0;
+
+    // eslint-disable-next-line no-plusplus
+    for (let i: number = 0; i < settings.length; i++) {
+        const entry: IRawThemeSetting = settings[i];
+
+        if (!entry.parameters) {
+            continue;
+        }
+
+        let scopes: string[];
+
+        if (typeof entry.scope === "string") {
+            let _scope: string = entry.scope;
+
+            // remove leading commas
+            _scope = _scope.replace(/^[,]+/, "");
+
+            // remove trailing commans
+            _scope = _scope.replace(/[,]+$/, "");
+
+            scopes = _scope.split(",");
+        } else if (Array.isArray(entry.scope)) {
+            scopes = entry.scope;
+        } else {
+            scopes = [""];
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        let parameters: Record<string, any> = {};
+
+        if (Boolean(entry.parameters) && typeof entry.parameters === "object") {
+            parameters = entry.parameters;
+        }
+
+        // eslint-disable-next-line no-plusplus
+        for (let j: number = 0; j < scopes.length; j++) {
+            const _scope: string = scopes[j].trim();
+
+            const segments: string[] = _scope.split(" ");
+
+            const scope: string = segments[segments.length - 1];
+            let parentScopes: string[] | undefined = undefined;
+
+            if (segments.length > 1) {
+                parentScopes = segments.slice(0, segments.length - 1);
+                parentScopes.reverse();
+            }
+
+            // eslint-disable-next-line no-plusplus
+            result[resultLen++] = new ParsedThemeRule(scope, parentScopes, i, parameters);
+        }
+    }
+
+    return result;
+}
+
+/**
+ * Resolve rules (i.e. inheritance).
+ */
+function resolveParsedThemeRules(parsedThemeRules: ParsedThemeRule[]): Theme {
+    // Sort rules lexicographically, and then by index if necessary
+    parsedThemeRules.sort((a: ParsedThemeRule, b: ParsedThemeRule) => {
+        let r: number = strcmp(a.scope, b.scope);
+
+        if (r !== 0) {
+            return r;
+        }
+
+        r = strArrCmp(a.parentScopes, b.parentScopes);
+
+        if (r !== 0) {
+            return r;
+        }
+
+        return a.index - b.index;
+    });
+
+    // Determine defaults
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let defaultParameters: Record<string, any> = {};
```

Comment:
Same here

Reply:
will fix the any with no-any patter but let's keep the record for consistency